enumerables.rb

```diff
@@ -0,0 +1,141 @@
+module Enumerable
+  def my_each
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index]
+      elsif is_a? Range
+        yield to_a[index]
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    (0...size).each do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    values = []
+    my_each do |val|
+      values.push(val) if yield(val)
+    end
+    values
+  end
+
+  def my_all?
+    if block_given?
+      values = true
+      my_each { |val| values = false unless yield(val) }
+      values
+    else
+      empty?
+    end
+  end
+
+  def my_any?
+    if block_given?
+      values = false
+      my_each { |val| values = true if yield(val) }
+      values
+    else
+      true
+    end
+  end
+
+  def my_none?
+    if block_given?
+      values = true
+      my_each { |val| values = false if yield(val) }
+      values
+    elsif self[0].nil? && !include?(true)
+      return true
+    else
+      return false
+    end
+    values
+  end
+
+  def my_count
+    count = 0
+    my_each do |num|
+      count += 1 if yield(num)
+    end
+    count
+  end
+
+  def my_map
+    mapped = []
+    my_each do |i|
+      mapped << (proc.nil? ? proc.call(i) : yield(i))
+    end
+    mapped
+  end
```

Comment:
- [ ] should have working `my_map` **(covered by automated tests)** identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should return an enumerator if no block or argument is given.